cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/StreamUtils.java

```diff
@@ -136,6 +136,25 @@ public static int getSequenceId(String name) {
   }
 
   /**
+   * Returns the writer instance number of the given file name.
+   *
+   * @param name Name of the file.
+   * @return The writer instance number of the stream file.
+   * @see StreamInputFormat
+   */
+  public static int getWriterInstanceId(String name) {
```

Comment:
Currently the logic is governed by the `LocationStreamFileWriterFactory`. Might need to see where is the best place to have this knowledge (seems like stream handler is the right place).